src/ImageSharp.Web/ImageSharpTagHelper.cs

```diff
@@ -0,0 +1,366 @@
+            }
+
+            if (this.PadColor.HasValue)
+            {
+                commands.Add(ResizeWebProcessor.Color, this.PadColor.Value.ToHex());
+            }
+
+            if (this.Compand.HasValue)
+            {
+                commands.Add(ResizeWebProcessor.Compand, this.Compand.Value.ToString(this.parserCulture));
+            }
+
+            if (this.Orient.HasValue)
+            {
+                commands.Add(ResizeWebProcessor.Orient, this.Orient.Value.ToString(this.parserCulture));
+            }
+
+            if (this.Sampler.HasValue)
+            {
+                commands.Add(ResizeWebProcessor.Sampler, this.Sampler.Value.Name);
+            }
+        }
+
+        private void AddAutoOrientCommands(CommandCollection commands)
+        {
+            if (this.AutoOrient.HasValue)
+            {
+                commands.Add(AutoOrientWebProcessor.AutoOrient, this.AutoOrient.Value.ToString());
+            }
+        }
+
+        private void AddFormatCommands(CommandCollection commands)
+        {
+            if (this.Format.HasValue)
+            {
+                commands.Add(FormatWebProcessor.Format, this.Format.Value.Name);
+            }
+        }
+
+        private void AddBackgroundColorCommands(CommandCollection commands)
+        {
+            if (this.BackgroundColor.HasValue)
+            {
+                commands.Add(BackgroundColorWebProcessor.Color, this.BackgroundColor.Value.ToHex());
+            }
+        }
+
+        private void AddQualityCommands(CommandCollection commands)
+        {
+            if (this.Quality.HasValue)
+            {
+                commands.Add(QualityWebProcessor.Quality, this.Quality.Value.ToString(this.parserCulture));
+            }
+        }
+
+        private static string AddQueryString(
+            ReadOnlySpan<char> uri,
+            CommandCollection commands)
+        {
+            ReadOnlySpan<char> uriToBeAppended = uri;
+            ReadOnlySpan<char> anchorText = default;
+
+            // If there is an anchor, then the query string must be inserted before its first occurrence.
+            int anchorIndex = uri.IndexOf('#');
+            if (anchorIndex != -1)
+            {
+                anchorText = uri.Slice(anchorIndex);
+                uriToBeAppended = uri.Slice(0, anchorIndex);
+            }
+
+            int queryIndex = uriToBeAppended.IndexOf('?');
+            bool hasQuery = queryIndex != -1;
+
+            StringBuilder sb = new();
+            sb.Append(uriToBeAppended);
+
+            foreach (KeyValuePair<string, string> parameter in commands)
+            {
+                if (parameter.Value is null)
+                {
+                    continue;
+                }
+
+                sb.Append(hasQuery ? '&' : '?');
+                sb.Append(UrlEncoder.Default.Encode(parameter.Key));
+                sb.Append('=');
+                sb.Append(UrlEncoder.Default.Encode(parameter.Value));
```

Comment:
I noticed all command keys are lowercase, but the values (especially for enums) are capitalized. Even though the commands are case-insensitive and work the same regardless, I think it's nicer (at least to the eye) to have all lowercase values as well:
```suggestion
                sb.Append(UrlEncoder.Default.Encode(parameter.Value.ToLowerInvariant()));
```

Reply:
It's also 2X more expensive.